### ISSUE TITLE
Change default branch to "main"

### ISSUE THREAD

**Issue by seand7565** [solidusio/solidus#3663]
The etymology of the name "master" for the default branch is [problematic](https://tools.ietf.org/id/draft-knodel-terminology-00.html), but more than that, it's just not very descriptive. 

There are a few alternatives, but as [this article](https://www.hanselman.com/blog/EasilyRenameYourGitDefaultBranchFromMasterToMain.aspx) describes, I think "main" fits much better. It's clearer as to what the purpose of the branch is:
![image](https://user-images.githubusercontent.com/5720486/84501116-8923c880-ac7b-11ea-9645-9710fd1ab528.png)

**vs.**

![image](https://user-images.githubusercontent.com/5720486/84501121-8b862280-ac7b-11ea-8853-59401360151e.png)

(even the appropriate definition of master is just the word "main", which to me is a clear indication that we should be using it instead)

I don't think this change is a huge one, and it gets us away from using problematic and unclear terminology. Hearsay on Twitter is that GitHub is considering making this change themselves, though I can't find anything official to back that up.

That being said, for those on cutting edge, this change will be breaking, so we should consider deprecating (if that's even possible?) master branch, or running both master and main branches as copies of each other until a large release like 3.0.

**Comment by stefano-sarioli** [comment]
I think that this is an issue we should consider, and fix.

GitHub is working on some updates to make this easier and painless, you can find more information [here](https://github.com/github/renaming#later-this-year). 
We should probably wait for the new features by GitHub to have avoid problems with the migration.

**Comment by jarednorman** [comment]
Yeah, that was my thought as well. I'm happy to adopt this new convention, but in the interest of not making extra work for ourselves we can wait until they've rolled out whatever tooling they're planning to make the transition easy/seamless.

**Comment by aldesantis** [comment]
If GitHub helps us make that transition smooth, I'm all in favor of this.

**Comment by ikraamg** [comment]
[Renaming](https://github.com/github/renaming#renaming-existing-branches) has been made possible a few months ago, and it looks pretty safe, but we should properly assess the practicality of the change.

This is the only [risk](https://docs.github.com/en/github/administering-a-repository/renaming-a-branch) that I read about: 

> Although file URLs are automatically redirected, raw file URLs are not redirected. GitHub does not perform any redirects if users perform a git pull for the previous branch name.

**Comment by seand7565** [comment]
Just as a heads up: Rails changed their `master` branch to `main` back in [February](https://github.com/rails/rails/pull/40254), and as far as I can tell it hasn't caused too much in the way of issues for developers. At least, I haven't heard any complaints since - TBH I wasn't even aware it changed until I looked just now.

**Comment by jarednorman** [comment]
Yeah, it's been pretty painless in my experience. I've seen a few projects switch to that new default now.

**Comment by waiting-for-dev** [comment]
I agree... and even if it could cause some small problem, I think that inclusivity is more important

**Comment by jarednorman** [comment]
I think the only annoying thing is we may need to update all the Gemfiles in contrib to use "main" as the default branch. It's currently hardcoded to "master".

**Comment by jarednorman** [comment]
Might need to update build matrixes too.
